crates/stdio_server/src/session/handler/on_move.rs

```diff
@@ -178,6 +178,14 @@ impl<'a> OnMoveHandler<'a> {
         match utility::read_preview_lines(path.as_ref(), lnum, self.size) {
             Ok((lines_iter, hi_lnum)) => {
                 let fname = format!("{}", path.as_ref().display());
+                let max_width = 2 * self.context.winwidth.unwrap_or(100) as usize;
+                let lines_iter = lines_iter.map(|line| {
+                    if line.len() > max_width {
+                        format!("{}......", &line[..max_width])
+                    } else {
+                        line
+                    }
+                });
```

Comment:
@ImmemorConsultrixContrarie Any insight to fix the code repeatation here?

Reply:
```rust
impl Whatever {
    fn preview_truncation<S: Borrow<str>>(&self, lines: impl Iterator<Item=S>) -> impl Iterator<Item=String> {
        let max_width = 2 * self.context.winwidth.unwrap_or(100) as usize;
        let lines_iter = lines_iter.map(move |line| {
            let line = line.borrow();
            if line.len() > max_width {
                format!("{}......", &line[..max_width])
            } else {
                line
            }
        });
    }
}
```

Reply:
`move` does the trick. 🎉

Reply:
Btw, It's better to change
`format!("{}......", &line[..max_width])`
into
```rust
line.truncate(max_width);
line.push_str("……");
line
```

to avoid unnecessary allocation here.

Reply:
Also, string index could panic if you would accidentally point into the middly of multibyte char. Probably one of `str` methods could help with this.

Reply:
Okay, it probably should be written as

```rust
line.truncate((0..max_width).find(|idx| line.is_char_boundary(idx)).unwrap());
line.push_str("……");
line
```

@liuchengxu

Reply:
This thing won't panic. However, for *multichar* sequences this could show a wrong thing, like showing 👩 or 🔬 emoji instead of 👩‍🔬 .

Reply:
Unicode strings are hard, yes.

Reply:
Yes, master, I'm on it...